### ISSUE TITLE
`callable` check does not excludes `None`

### ISSUE THREAD

**Issue by pawelrubin** [microsoft/pyright#1065]
**Describe the bug**
When checking the type of a variable via `callable`, PyRight will still see the variable as Optional.

**To Reproduce**
Consider this piece of code
```python
from typing import Callable, Optional

Function = Callable[[int], int]


def bar(f: Function):
    pass


def foo(arg: Optional[Function]) -> None:
    if callable(arg):
        bar(arg)  # Type "None" cannot be assigned to type "Function"
```
**Expected behavior**
Since `None` is not callable, there should be no issues.

**Additional context**
Pyright extensions - 1.1.74
Python - 3.8.3

**Comment by erictraut** [comment]
Thanks for the bug report. Yes, I missed this case in the type narrowing logic for `callable`. It will be fixed in the next release of Pyright and Pylance.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.75, which I just published. It will also be included in the next version of Pylance.
